universal_transfer_operator/src/datasets/base.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from urllib.parse import urlparse
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+from attr import define, field
+
+try:
+    # Airflow >= 2.4
+    from airflow.datasets import Dataset
+
+    DATASET_SUPPORT = True
+except ImportError:
+    # Airflow < 2.4
+    Dataset = object
+    DATASET_SUPPORT = False
+
+
+@define
+class UniversalDataset(LoggingMixin, Dataset):
+    """
+    Repersents all file dataset, and abstract away the details related to location and file types.
+    Intended to be used within library.
+
+    :param path: Path to a file in the filesystem/Object stores
+    :param conn_id: Airflow connection ID
+    """
+
+    path: str
```

Comment:
@sunank200 I thought again about this `path`. I think it is not a good idea to define the absolute path here. Can't we use the connection details to construct the path for operators? So we only require a table name for tables for example as we do in the astro-sdk when converting path to dataset.

Reply:
I am a bit worried about users defining database host in the path such as snowflake. In case of s3 it is not so much of a deal, because you dont authorize against the bucket. But for databases, in my opinion, I see the host as sensitive information.

Reply:
+1

Reply:
Removed it: https://github.com/astronomer/astro-sdk/pull/1492/commits/b59f087700a88b7f24989d0619c89bd06771c8e5